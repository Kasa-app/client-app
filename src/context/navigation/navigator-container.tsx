import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LoginPage from '../../screens/authentification/login/login-page-view';
import SignUpPage from '../../screens/authentification/signup/signup-page';
import StartupPage from '../../screens/authentification/startup/startup-page';
import ServiceDetailsPage from '../../screens/service-details/service-details-page';
import { TabsNavigation } from '../../components/Bottom-Navigation-View/Tabs-navigation/Bottom/tabs-navigation-view';
import ServiceList from '../../screens/service-list/service-list-page';
import SearchJobbersPage from '../../screens/search-jobbers/search-jobbers';
import JobberProfilePage from '../../screens/jobber-profile/jobber-profile';
import BookingPage from '../../screens/booking/booking-page';
import LaunchPage from '../../screens/launch-page/launch-page';
import Chat from '../../components/Bottom-Navigation-View/Chat/Chat-View/chat';

const Stack = createNativeStackNavigator();

export const NavigatorContainer = () => {
    return (
        <NavigationContainer>
            <Stack.Navigator
                screenOptions={{
                    headerShown: false,
                    presentation: 'modal',
                    animationTypeForReplace: 'push',
                    animation: 'slide_from_bottom',
                    animationDuration: 200
                }}
                initialRouteName='LaunchScreen'
            >
                <Stack.Screen name="LaunchScreen" component={LaunchPage} />
                <Stack.Screen name="StartupScreen" component={StartupPage} />
                <Stack.Screen name="LoginScreen" component={LoginPage} />
                <Stack.Screen name="SignUpScreen" component={SignUpPage} />
                <Stack.Screen name="ServiceDetailsScreen" component={ServiceDetailsPage} />
                <Stack.Screen name="BookingScreen" component={BookingPage} />
                <Stack.Screen name="ChatScreen" component={Chat} />
                <Stack.Screen name="TabsNavigationScreen" component={TabsNavigation} />
                <Stack.Screen name="ServiceListScreen" component={ServiceList} />
                <Stack.Screen name="SearchJobbersScreen" component={SearchJobbersPage} />
                <Stack.Screen name="JobberProfileScreen" component={JobberProfilePage} />
            </Stack.Navigator>
        </NavigationContainer>
    );
};
