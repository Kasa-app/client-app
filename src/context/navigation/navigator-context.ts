import { createContext } from "react";
import { NavigatorContextType } from "../../models/contextType/NavigatorContextType";

export const NavigatorContext = createContext<NavigatorContextType>({
  activeStep: null,
  setActiveStep: () => { },
});