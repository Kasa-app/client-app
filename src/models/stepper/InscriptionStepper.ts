export enum InscriptionStepper {
  SignUpWithEmail = 'signUpWithEmail',
  UserEmail = 'User Email',
  UserName = 'User Name',
  UserBirthdate = 'User birthdate',
  UserPassword = 'User password',
}