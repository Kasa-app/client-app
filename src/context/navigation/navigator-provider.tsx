import React, { useState } from 'react';
import { NavigatorContext } from './navigator-context';
import { InscriptionStepper } from '../../models/stepper/InscriptionStepper';
import { NavigatorContextType } from '../../models/contextType/NavigatorContextType';

export const NavigatorProvider = ({ children }: any) => {
  const [activeStep, setActiveStep] = useState(InscriptionStepper.SignUpWithEmail);
  const context: NavigatorContextType =
  {
    activeStep,
    setActiveStep,
  };

  return (
    <NavigatorContext.Provider value={context}>
      {children}
    </NavigatorContext.Provider>
  );
};

