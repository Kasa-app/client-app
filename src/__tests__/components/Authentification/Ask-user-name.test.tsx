import { Text, TextInput } from 'react-native';
import React from 'react';
import { it } from '@jest/globals';
import { create, act } from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import { fireEvent } from 'react-native-testing-library';
import AskUserName from '../../../components/Authentification/Ask-user-name/ask-user-name';
import { NavigatorContext } from '../../../context/navigation/navigator-context';
import { InscriptionStepper } from '../../../models/stepper/InscriptionStepper';
import ButtonWithBackgroundColor from '../../../components/Button-with-color-background/button';
jest.mock('react-native-keyboard-aware-scroll-view', () => {
  return {
    KeyboardAwareScrollView: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  };
});

const mockNavigatorContextValue = {
  routeName: '',
  setRouteName: jest.fn(),
  activeStep: InscriptionStepper.UserName,
  setActiveStep: jest.fn(),
};
const MockNavigatorContextProvider = ({ children }: any) => (
  <NavigatorContext.Provider value={mockNavigatorContextValue}>
      {children}
  </NavigatorContext.Provider>
);

describe('AskUserName', () => {

  let tree: any

  beforeEach(() => {
    tree = render(<AskUserName />);
  });

  it('snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should call activeStep with userBirthDate when continue btn is pressed', () => {
    const component = render(<AskUserName />, { wrapper: MockNavigatorContextProvider });
    const button = component.root.findByType(ButtonWithBackgroundColor).props;

    act(() => button.handlePress());
    expect(mockNavigatorContextValue.setActiveStep).toHaveBeenCalledWith(InscriptionStepper.UserBirthdate);
  });

  it('when firstName is empty and lastName is not should disable continue btn', () => {
    const firstName = tree.root.findAllByType(TextInput)[0];
    const lastName = tree.root.findAllByType(TextInput)[1];

     fireEvent.changeText(firstName, '');
     fireEvent.changeText(lastName, 'test');

      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      expect(continueBtn.shouldDisable).toBe(true);
});

it('when lastName is empty and firstName is not should disable continue btn', () => {

    const firstName = tree.root.findAllByType(TextInput)[0];
    const lastName = tree.root.findAllByType(TextInput)[1];

     fireEvent.changeText(lastName, '');
     fireEvent.changeText(firstName, 'test');

      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      expect(continueBtn.shouldDisable).toBe(true);
});

it('when lastName and firstName are empty should disable continue btn', () => {

    const firstName = tree.root.findAllByType(TextInput)[0];
    const lastName = tree.root.findAllByType(TextInput)[1];

     fireEvent.changeText(lastName, '');
     fireEvent.changeText(firstName, '');

      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      expect(continueBtn.shouldDisable).toBe(true);
});

it('when lastName and firstName are not empty should enable continue btn', () => {

    const firstName = tree.root.findAllByType(TextInput)[0];
    const lastName = tree.root.findAllByType(TextInput)[1];

     fireEvent.changeText(lastName, 'test');
     fireEvent.changeText(firstName, 'test');

      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      expect(continueBtn.shouldDisable).toBe(false);
});


});