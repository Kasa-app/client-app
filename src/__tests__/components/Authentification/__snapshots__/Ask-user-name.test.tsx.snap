// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskUserName snapshot 1`] = `
<div>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": "10%",
        }
      }
    >
      <View>
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#E67918",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Nom"
              secureTextEntry={false}
              style={
                {
                  "fontSize": 13,
                  "height": "85%",
                  "paddingStart": "5%",
                  "width": "88%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginTop": "3%",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#E67918",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="PrÃ©nom"
              secureTextEntry={false}
              style={
                {
                  "fontSize": 13,
                  "height": "85%",
                  "paddingStart": "5%",
                  "width": "88%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginTop": "25%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#E2B288",
                "borderRadius": 4,
                "display": "flex",
                "height": undefined,
                "justifyContent": "center",
                "minWidth": "35%",
                "opacity": 1,
                "paddingVertical": 7,
                "width": "77%",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "black",
                    "fontFamily": "AvenirNext-Regular",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  {
                    "color": "gray",
                  },
                  undefined,
                ]
              }
            >
              Continuer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</div>
`;
